provisioner/powershell/provisioner.go

```diff
@@ -285,9 +286,9 @@ func (p *Provisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packe
 		if err != nil {
 			return fmt.Errorf("Error stating powershell script: %s", err)
 		}
-		if strings.HasSuffix(p.config.RemotePath, `\`) {
+		if os.IsPathSeparator(p.config.RemotePath[len(p.config.RemotePath)-1]) {
 			// path is a directory
-			p.config.RemotePath += filepath.Base((fi).Name())
+			p.config.RemotePath += filepath.Base(fi.Name())
```

Comment:
`os.IsPathSeparator` on Windows will check for `\` and `/` ! `filepath.Base` internally uses `IsPathSeparator` to find the base path, so it should be fine.